base/array.jl

```diff
@@ -383,6 +383,9 @@ function setindex!{T}(A::Array{T}, X::Array{T}, c::Colon)
     return A
 end
 
+setindex!{T}(A::Array{T}, x::Number, c::Colon) = fill!(A, x)
```

Comment:
Note that the `{T}` is not necessary here.

What is the problem with just doing `setindex!(A::AbstractArray, x::Number, c::Colon) = fill!(A, x)`?